src/org/zaproxy/zap/extension/ascanrulesAlpha/ApacheRangeHeaderDos.java

```diff
@@ -0,0 +1,184 @@
+	}
+
+	@Override
+	public String getDescription() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "desc");
+	}
+
+	@Override
+	public String getSolution() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "soln");
+	}
+
+	@Override
+	public String getReference() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "refs");
+	}
+
+	@Override
+	public String[] getDependency() {
+		return null;
+	}
+
+	@Override
+	public boolean targets(TechSet technologies) {
+		return technologies.includes(Tech.Apache);
+	}
+
+	@Override
+	public int getCategory() {
+		return Category.SERVER;
+	}
+
+	@Override
+	public int getRisk() {
+		return Alert.RISK_MEDIUM;
+	}
+
+	@Override
+	public int getCweId() {
+		return 400; // CWE-400: Uncontrolled Resource Consumption
+	}
+
+	@Override
+	public int getWascId() {
+		return 10; // WASC-10: Denial of Service
+	}
+
+	@Override
+	public void init() {
+
+	}
+
+	@Override
+	public void scan() {
+
+		// Check if the user stopped things. One request per URL so check before
+		// sending the request
+		if (isStop()) {
+			if (LOG.isDebugEnabled()) {
+				LOG.debug("Scanner " + getName() + " Stopping.");
+			}
+			return;
+		}
+
+		if (acceptsRangeRequests()) {// Server handles ranges
+			HttpMessage newRequest = getNewMsg();
+			newRequest = setRequestHeaders(newRequest, "3-0,1-1,2-2,3-3,4-4,5-5,6-6,7-7,8-8,9-9,10-10,11-11");
+			// Send a request for 11 ranges, 1 more than permitted
+			try {
+				sendAndReceive(newRequest, false);
+			} catch (IOException e) {
+				LOG.warn("An error occurred while checking [" + newRequest.getRequestHeader().getMethod() + "] ["
+						+ newRequest.getRequestHeader().getURI() + "] for Apache Range Header DoS (CVE-2011-3192)."
+						+ "Caught " + e.getClass().getName() + " " + e.getMessage());
+				return;
+			}
+
+			if (newRequest.getResponseHeader().getStatusCode() == HttpStatusCode.PARTIAL_CONTENT) {// 206
+				bingo(getRisk(), // Risk
+						Alert.CONFIDENCE_MEDIUM, // Confidence/Reliability
+						getName(), // Name
+						getDescription(), // Description
+						newRequest.getRequestHeader().getURI().toString(), // URI
+						null, // Param
+						"", // Attack
+						"", // OtherInfo
+						getSolution(), // Solution
+						"", // Evidence
+						getCweId(), // CWE ID
+						getWascId(), // WASC ID
+						newRequest); // HTTPMessage
+			}
+		}
+	}
+
+	private boolean acceptsRangeRequests() {
+		HttpMessage chkRequest = getNewMsg();
+		chkRequest = setRequestHeaders(chkRequest, "0-6");
+
+		try {
+			sendAndReceive(chkRequest, false);
+		} catch (IOException e) {
+			LOG.warn("An error occurred while validating [" + chkRequest.getRequestHeader().getMethod() + "] ["
+					+ chkRequest.getRequestHeader().getURI()
+					+ "] for Apache Range Header DoS (CVE-2011-3192) applicability." + "Caught "
+					+ e.getClass().getName() + " " + e.getMessage());
+		}
+
+		if (chkRequest.getResponseHeader().getStatusCode() == HttpStatusCode.PARTIAL_CONTENT) {// 206

```

Comment:
Could return the value of the comparison instead.
The comment could be removed?